libmscore/measure.cpp

```diff
@@ -2147,6 +2147,84 @@ void Measure::read(XmlReader& e, int staffIdx)
       }
 
 //---------------------------------------------------------
+//   checkMeasure
+//    after opening / paste and every read operation
+//    this method checks for gaps and fills them
+//    with invisible rests
+//---------------------------------------------------------
+
+void Measure::checkMeasue(int staffIdx)
+      {
+      for (int track = staffIdx * VOICES + 1; track % VOICES && hasVoice(track); track++) {
+            Segment* seg = first();
+            if (!seg->element(track))
+                  seg = seg->nextCR(track, false);
+            if (!seg->element(track))
+                  continue;
+
+            Segment* pseg = 0;
+            int stick = tick();
+            int ticks = seg->tick() - stick;
+
+            while (seg) {
+                  if (ticks) {
+                        Fraction f = Fraction::fromTicks(ticks);
+                        std::vector<TDuration> dList = toDurationList(f, true);
+                        if (dList.empty())
+                              break;
+
+                        for (const TDuration& d : dList) {
+                              Segment* segment = 0;
+                              Rest* rest = new Rest(score());
+                              rest->setDurationType(d);
+                              rest->setDuration(d.fraction());
+                              rest->setTrack(track);
+                              segment = getSegment(rest, stick);
+                              segment->add(rest);
+                              rest->setTrulyInvisible(true);
+                              seg = segment;
+                              }
+                        }
```

Comment:
It might be beneficial to allow arbitrary duration values for trulyInvisible rests and remove the call to toDurationList(...).